smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServerGenerator.java

```diff
@@ -46,114 +46,202 @@ static void generateServiceHandler(SymbolProvider symbolProvider,
+                            + "serdeContextBase);");
                 });
                 writer.openBlock("switch (target.operation) {", "}", () -> {
                     for (OperationShape operation : operations) {
-                        generateHandlerCase(writer, operation, symbolProvider.toSymbol(operation));
+                        String opName = operation.getId().getName();
+                        writer.openBlock("case $S : {", "}", opName, () -> {
+                            writer.write("return handle(request, this.serializerFactory($S), this.service.$L, "
+                                    + "this.serializeFrameworkException);",
+                                    opName, symbolProvider.toSymbol(operation).getName());
+                        });
                     }
                 });
             });
         });
     }
 
-    private static void generateHandlerCase(TypeScriptWriter writer,
-                                            Shape operationShape,
-                                            Symbol operationSymbol) {
-        String opName = operationShape.getId().getName();
-        writer.openBlock("case $S : {", "}", opName, () -> {
-            writer.write("let serializer = this.serializerFactory($S);", opName);
-            writer.write("let input;");
-            writer.openBlock("try {", "} catch (error: unknown) {", () -> {
-                writer.openBlock("input = await serializer.deserialize(request, {", "});", () -> {
-                    writer.write("endpoint: () => Promise.resolve(request), ...this.serdeContextBase");
-                });
+    static void generateOperationHandler(SymbolProvider symbolProvider,
+                                                 Shape serviceShape,
+                                                 OperationShape operation,
+                                                 TypeScriptWriter writer) {
+        addCommonHandlerImports(writer);
+
+        writeSerdeContextBase(writer);
+        writeHandleFunction(writer);
+
+        String operationName = operation.getId().getName();
+        Symbol serviceSymbol = symbolProvider.toSymbol(serviceShape);
+        Symbol operationSymbol = symbolProvider.toSymbol(operation);
+
+        Symbol inputSymbol = operationSymbol.expectProperty("inputType", Symbol.class);
+        Symbol outputSymbol = operationSymbol.expectProperty("outputType", Symbol.class);
+        Symbol handlerSymbol = operationSymbol.expectProperty("handler", Symbol.class);
+
+        writer.openBlock("export class $L implements __ServiceHandler {", "}", handlerSymbol.getName(), () -> {
+            writer.write("private operation: __Operation<$T, $T>;", inputSymbol, outputSymbol);
+            writer.write("private mux: __Mux<$S, $S>;", serviceShape.getId().getName(), operationName);
+            writer.write("private serializer: __OperationSerializer<$T, $S, __SmithyException>;",
```

Comment:
This should use the operation's error type

Reply:
Yep, will fix.